File: 883-car-fleet/car-fleet.ts
interface Car {
    speed: number
    position: number
}

function carFleet(target: number, position: number[], speed: number[]): number {
    const stack: number[] = []
    const cars: Car[] = position.map((pos, index) => ({ position: pos, speed: speed[index] }))

    cars.sort((a, b) => b.position - a.position)

    for (const car of cars) {
        const time: number = (target - car.position) / car.speed

        if (stack.length === 0 || time > stack[stack.length-1]) stack.push(time)
    }

    return stack.length
};

File: 883-car-fleet/car-fleet.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        cars = sorted(zip(position, speed), reverse=True)
        print(cars)

        stack = []

        for pos, spd in cars:
            time = (target - pos) / spd

            if not stack or time > stack[-1]:
                stack.append(time)

        return len(stack)